sdk/storage/AzureStorageBlob/Source/Data/BlobTransfer+CoreDataClass.swift

```diff
@@ -83,6 +83,58 @@ public class BlobTransfer: NSManagedObject, TransferImpl {
         }
     }
 
+    // FIXME: The fact that BlobProperties is a struct causes serious problems here. It will always return nil if
+    // allowed to be optional.
+    internal var properties: BlobProperties {
+        get {
+            guard let propertiesString = rawProperties,
+                let jsonData = propertiesString.data(using: .utf8) else { return BlobProperties() }
+            do {
+                return try StorageJSONDecoder().decode(BlobProperties.self, from: jsonData)
+            } catch {
+                return BlobProperties()
+            }
+        }
+
+        set {
+            if let newJson = try? StorageJSONEncoder().encode(newValue) {
+                rawProperties = String(data: newJson, encoding: .utf8)
+            }
+        }
+    }
+
+    internal var uploadOptions: UploadBlobOptions? {
+        get {
+            guard transferType == .upload,
+                let options = rawOptions,
+                let jsonData = options.data(using: .utf8) else { return nil }
+            return try? StorageJSONDecoder().decode(UploadBlobOptions.self, from: jsonData)
```

Comment:
Do you think we should memoize this, so that multiple `get`s on this property don't need to invoke the JSON parser more than once? Likewise we could optimize the `set` the same way so that if the new value is equal to the existing one, we don't invoke the JSON serializer.

Reply:
Sounds smart to me!

Reply:
Done.